geode-core/src/main/java/org/apache/geode/management/internal/operation/RebalanceOperationPerformer.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.management.internal.operation;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.geode.annotations.Experimental;
+import org.apache.geode.cache.Cache;
+import org.apache.geode.management.cli.Result;
+import org.apache.geode.management.internal.cli.commands.RebalanceCommand;
+import org.apache.geode.management.internal.cli.result.model.InfoResultModel;
+import org.apache.geode.management.internal.cli.result.model.ResultModel;
+import org.apache.geode.management.internal.cli.result.model.TabularResultModel;
+import org.apache.geode.management.operation.RebalanceOperation;
+import org.apache.geode.management.runtime.RebalanceResult;
+
+@Experimental
+public class RebalanceOperationPerformer {
+  public static RebalanceResult perform(Cache cache, RebalanceOperation parameters) {
+    try {
+      RebalanceCommand rebalanceCommand = new RebalanceCommand();
+      rebalanceCommand.setCache(cache);
+
+      ResultModel result = rebalanceCommand
```

Comment:
I would like to see us not dependent on the command implementation at all. I believe the command also starts up threads to do the real work, and this performer is already in a thread. Please try to see how much work it is to do the real work in the performer instead of delegating to the command. If it's too much, you can do it in a separate PR.

Reply:
Do you want to use RebalanceCommand.ExecuteRebalanceWithTimeout directly?  Or go even deeper and invoke `executeFunction` directly?

Reply:
That ExecuteRebalanceWithTimeout.executeRebalanceWithTimeout looks like it's doing the real thing. There is even comments there saying we need to pull it out to a separate class. We should pull that out and have both the command and performer call that.

Reply:
@jinmeiliao was it just the extra thread you wanted to avoid on the RebalanceCommand or did you want avoid other things in the command also? For example is it good for us to get a ResultModel back and translate it into the json response? It seems like a good idea to reuse as much of this code as possible since it is already tested. Is this true?

Reply:
ResultModel and Command are "gfsh" concept, I would love to see CMS not using any Gfsh construct at all, since eventually we want to get rid of gfsh v1. In my mind, I would love to see a RebalanceResult that would have those attributes like "regionRebalanced", "TotalBucketCreateBytes",  "TotalBucketCreatesCompleted" etc as fields in this RebalanceResult, and both gfsh command would call that to construct its own ResultModel. But looks like its going to too much work.

Reply:
Right now a lot of the implementation of the inner class `RebalanceCommand.ExecuteRebalanceWithTimeout` still depends on helper methods in the parent class `GfshCommand`.  For today I will refactor just enough to get the callable without incurring the overhead of an extra thread, then tomorrow we can look into teasing out ExecuteRebalanceWithTimeout into a standalone class.

Reply:
actually I don't mind you put all this work in another PR, as long as we have this work to be done captured in another ticket.

Reply:
It makes a difference in the output structure (the rebalance result), so releasing rebalance with one format then totally changing it may be less than desirable.  If it's not too much work to leverage the existing org.apache.geode.cache.control.RebalanceResults interface (would require moving it from geode-core to geode-management) that might be desirable.

Reply:
Tried today to remove dependency on gfsh / ResultModel, but it's much more entangled than I realized.  That will have to be a project for another PR.